inference-engine/tests/functional/plugin/shared/src/subgraph_tests/basic_lstm.cpp

```diff
@@ -0,0 +1,214 @@
+
+    const size_t hidden_size = 118;
+    const size_t batch_size = 1;
+
+    outPrc = InferenceEngine::Precision::FP32;
+
+    //Reshape_1 [1,490] -> [1, 10, 49]
+    std::vector<uint64_t> outFormShapes1 = { batch_size, 10, 49 };
+    auto pattern1 = std::make_shared<ngraph::opset1::Constant>(ngraph::element::Type_t::i64, ngraph::Shape{3}, outFormShapes1);
+    auto reshape1 = std::make_shared<ngraph::opset1::Reshape>(params[0], pattern1, false);
+
+    auto reshape1_shape = reshape1->output(0).get_shape();
+    auto H_init = ngraph::builder::makeConstant(ngPrc, { batch_size, hidden_size }, {}, true);
+    auto C_init = ngraph::builder::makeConstant(ngPrc, { batch_size, hidden_size }, {}, true);
+
+    auto H_t = std::make_shared<ngraph::opset1::Parameter>(ngPrc, ngraph::Shape{ batch_size, hidden_size });
+    auto C_t = std::make_shared<ngraph::opset1::Parameter>(ngPrc, ngraph::Shape{ batch_size, hidden_size });
+
+    //Body
+    auto X = std::make_shared<ngraph::opset1::Parameter>(ngPrc, ngraph::Shape{ batch_size, 1, reshape1_shape[2] });
+    auto weightsNode = ngraph::builder::makeConstant(ngPrc, { 4 * hidden_size, reshape1_shape[2] }, {}, true);
+    auto reccurrenceWeightsNode = ngraph::builder::makeConstant(ngPrc, { 4 * hidden_size, hidden_size }, {}, true);
+
+    //lstm [1, 10], [1, 118], [1, 118] -> [1, 118], [1, 118]
+    outFormShapes1 = { batch_size, reshape1_shape[2] };
+    auto constantX = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{2}, outFormShapes1);
+    auto lstm1 = std::make_shared<ngraph::opset1::LSTMCell>(std::make_shared<ngraph::opset1::Reshape>(X, constantX, false),
+        H_t, C_t,
+        weightsNode, reccurrenceWeightsNode, hidden_size);
+
+    auto H_o = lstm1->output(0);
+    auto C_o = lstm1->output(1);
+
+    //TensorIterator [1, 10, 49] [1, 118], [1, 118] -> [1, 118]
+    auto body = std::make_shared<ngraph::opset1::TensorIterator::BodyLambda>(
+        ngraph::OutputVector{ H_o, C_o }, ngraph::ParameterVector{ X, H_t, C_t });
+
+    auto tensor_iterator = std::make_shared<ngraph::opset1::TensorIterator>();
+    tensor_iterator->set_body(body);
+
+    //input tensor shape: [1, 10, 49] chunk shape: [1, 1, 49]
+    tensor_iterator->set_sliced_input(X, reshape1, 0, 1, 1, -1, 1);
+    tensor_iterator->set_merged_input(H_t, H_init, H_o);
+    tensor_iterator->set_merged_input(C_t, C_init, C_o);
+
+    auto out0 = tensor_iterator->get_iter_value(H_o, -1);
+
+    const size_t output_size = 12;
+    auto fc1 = ngraph::builder::makeFullyConnected(out0, ngPrc, output_size, true, { hidden_size, output_size }, { 1 }, { 1 });
+
+    ngraph::ResultVector results {std::make_shared<ngraph::opset1::Result>(fc1)};
+    function = std::make_shared<ngraph::Function>(results, params, "Basic_LSTM_S");
+}
+
+void Basic_LSTM_S::Run() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    ConfigurePlugin();
+    LoadNetwork();
+    Infer();
+
+    const auto& actualOutputs = GetOutputs();
```

Comment:
Looks like you could just override CalculateRefs() method instead. But it's a general comment to make it a little bit more elegant, I'm not against keeping it as is.